tree/splitter.h

```diff
@@ -0,0 +1,43 @@
+#ifndef _TREE_SPLITTER_H__
+#define _TREE_SPLITTER_H__
+
+/**
+ * Information regarding the split of a given node.
+ * This will be used by the tree during its construction.
+ */
+struct SplitterRecord
```

Comment:
Could we move this somewhere else? We may need to change this structure depending on the concrete `Splitter` class.